mergify_engine/web.py

```diff
@@ -151,6 +160,67 @@ def subscription_cache(installation_id):  # pragma: no cover
     return "Cache cleaned", 200
 
 
+SimulatorSchema = voluptuous.Schema({
+    voluptuous.Required('url'): voluptuous.Url(),
+    voluptuous.Required('mergify.yml'): str,
+})
+
+
+def abort(message, code=400):
```

Comment:
You could rather use something like:

```python
@app.errorhandler(exceptions.MergifyException)
def handle_mergify_exception(error):
    response = flask.jsonify({
        "error": error.__class__.__name__,
        "message": error.message,
        "detail": error.detail,
    })
    response.status_code = error.status_code
    return response
```

And define/raise proper exceptions.

Reply:
This give me the idea to rework a bit the rule validator to remove most of the manual 400 reporting. I just put a handler for mashmallow.Invalid.